server/ladder_service.py

```diff
@@ -125,64 +151,77 @@ def _cancel_existing_searches(self, initiator: Player) -> List[Search]:
-
-        mapname = map_path[5:-4]
-        # FIXME: Database filenames contain the maps/ prefix and .zip suffix.
-        # Really in the future, just send a better description
-        self._logger.debug("Starting ladder game: %s", game)
-        await host.lobby_connection.launch_game(game, is_host=True, use_map=mapname)
-        try:
-            hosted = await game.await_hosted()
-            if not hosted:
-                raise TimeoutError("Host left lobby")
-        except TimeoutError:
-            msg = {"command": "game_launch_timeout"}
-            await asyncio.gather(
-                host.lobby_connection.send(msg),
-                guest.lobby_connection.send(msg)
+            host.game = game
+            guest.game = game
+
+            game.map_file_path = map_path
+
+            game.set_player_option(host.id, 'StartSpot', 1)
+            game.set_player_option(guest.id, 'StartSpot', 2)
+            game.set_player_option(host.id, 'Faction', host.faction)
+            game.set_player_option(guest.id, 'Faction', guest.faction)
+            game.set_player_option(host.id, 'Color', 1)
+            game.set_player_option(guest.id, 'Color', 2)
+            game.set_player_option(host.id, 'Army', 2)
+            game.set_player_option(guest.id, 'Army', 3)
+
+            # Remembering that "Team 1" corresponds to "-": the non-team.
+            game.set_player_option(host.id, 'Team', 1)
+            game.set_player_option(guest.id, 'Team', 1)
+
+            mapname = map_path[5:-4]
+            # FIXME: Database filenames contain the maps/ prefix and .zip suffix.
+            # Really in the future, just send a better description
+            self._logger.debug("Starting ladder game: %s", game)
+            await host.lobby_connection.launch_game(game, is_host=True, use_map=mapname)
+            try:
+                hosted = await game.await_hosted()
+                if not hosted:
+                    raise TimeoutError("Host left lobby")
+            except TimeoutError:
+                msg = {"command": "game_launch_cancelled"}
+                with contextlib.suppress(DisconnectedError):
+                    # Sending to host might not be needed, but we do it anyways
+                    await asyncio.gather(
+                        host.send_message(msg),
+                        guest.send_message(msg)
+                    )
+                # TODO: Uncomment this line once the client supports `game_launch_cancelled`.
+                # Until then, returning here will cause the client to think it is
+                # searching for ladder, even though the server has already removed it
+                # from the queue.
+
+                # return
+                self._logger.debug("Ladder game failed to launch due to a timeout")
+
+            # TODO: Graceful handling of NoneType errors due to disconnect
+            await guest.lobby_connection.launch_game(
+                game, is_host=False, use_map=mapname
```

Comment:
Do we need to launch game for `guest` if game is not hosted?

Reply:
Yes, let me paste my comment about it here:
```
# TODO: Uncomment this line once the client supports `game_launch_cancelled`.
# Until then, returning here will cause the client to think it is
# searching for ladder, even though the server has already removed it
# from the queue.
```

This is how the client expects messages to happen (and how they would have been handled before). The other option is of course to purposefully introduce a bug to force the client guys to fix it properly, but that's probably not the best idea.